library/Gallery.js

```diff
@@ -219,25 +220,47 @@ export default class Gallery extends Component {
   }
 
   renderPage(pageData, pageId, layout) {
-    const { onViewTransformed, onTransformGestureReleased, ...other } = this.props;
+    const {
+      onViewTransformed, onTransformGestureReleased, renderLoading, renderError,
+      ...other
+    } = this.props;
+
     return (
-      <Image
-        {...other}
-        onViewTransformed={((transform) => {
-           onViewTransformed && onViewTransformed(transform, pageId);
-        }).bind(this)}
-        onTransformGestureReleased={((transform) => {
-           onTransformGestureReleased && onTransformGestureReleased(transform, pageId);
-        }).bind(this)}
-        ref={((ref) => {
-           this.imageRefs.set(pageId, ref);
-        }).bind(this)}
-        key={'innerImage#' + pageId}
-        style={{width: layout.width, height: layout.height}}
-        source={{uri: pageData}}/>
+      <View>
+        {renderLoading && !this.state.status[pageId] &&
+          <View style={Style.underlay}>{renderLoading()}</View>}
+
+        {renderError && this.state.status[pageId] === 'error' &&
+          <View style={Style.underlay}>{renderError()}</View>}
+
+        <Image
+          {...other}
+          onViewTransformed={((transform) => {
+             onViewTransformed && onViewTransformed(transform, pageId);
+          }).bind(this)}
+          onTransformGestureReleased={((transform) => {
+             onTransformGestureReleased && onTransformGestureReleased(transform, pageId);
+          }).bind(this)}
+          ref={((ref) => {
+             this.imageRefs.set(pageId, ref);
+          }).bind(this)}
+          key={'innerImage#' + pageId}
+          style={{width: layout.width, height: layout.height}}
+          source={{uri: pageData}}
+          onLoad={this.setLoadedStatus.bind(this, pageId)}
+          onError={this.setErrorStatus.bind(this, pageId)}/>
```

Comment:
It would be great to add a listener on `onProgress`, that way you could even call `renderLoading` with the progress. See [react-native-image-progress](https://github.com/oblador/react-native-image-progress/blob/master/index.js#L94-L106)

Reply:
Could you look at:
https://github.com/archriss/react-native-image-gallery/pull/13

p. 5.

This way you can achieve whatever you want with a custom Image component.

It doesn't look like @ldn0x7dc is maintaining this repo and it's dependencies.